swap/src/bin/swap.rs

```diff
@@ -94,23 +94,53 @@ async fn main() -> Result<()> {
                 EventLoop::new(swap_id, swarm, alice_peer_id, bitcoin_wallet.clone())?;
             let event_loop = tokio::spawn(event_loop.run());
 
-            let send_bitcoin = determine_btc_to_swap(
-                event_loop_handle.request_quote(),
+            debug!("Requesting quote");
+            let bid_quote = event_loop_handle.request_quote().await?;
+            info!("Received quote: 1 XMR ~ {}", bid_quote.price);
+
+            let initial_max_giveable = bitcoin_wallet.max_giveable(TxLock::script_size()).await?;
+            let (send_bitcoin, fees) = determine_btc_to_swap(
+                bid_quote.clone(),
                 bitcoin_wallet.balance(),
                 bitcoin_wallet.new_address(),
                 async {
```

Comment:
>  If you can come up with a clever way to re-arrange code in/around determine_btc_to_swap let me know please :)

You just need to decompose this future into multiple so you can move the loop inside. I also think you can simplify this implementation quite a bit by going into the direction of what @bonomat was doing in https://github.com/comit-network/xmr-btc-swap/pull/474:

Include relevant information as fields:

```rust
tracing::info!(
    %balance,
    %max_giveable,
	%min_buy,
    "Received BTC",
);
```

Also, I believe you will need to `sync` in the loop, otherwise you never learn about your new balance.